src/java.base/share/classes/java/util/Properties.java

```diff
@@ -814,18 +814,20 @@ public void save(OutputStream out, String comments)  {
      * If the comments argument is not null, then an ASCII {@code #}
      * character, the comments string, and a line separator are first written
      * to the output stream. Thus, the {@code comments} can serve as an
-     * identifying comment. Any one of a line feed ('\n'), a carriage
-     * return ('\r'), or a carriage return followed immediately by a line feed
-     * in comments is replaced by a line separator generated by the {@code Writer}
-     * and if the next character in comments is not character {@code #} or
-     * character {@code !} then an ASCII {@code #} is written out
-     * after that line separator.
+     * identifying comment. Any one of a line feed ({@code \n}), a carriage
+     * return ({@code \r}), or a carriage return followed immediately by a line feed
+     * ({@code \r\n}) in comments is replaced by a
+     * {@link System#lineSeparator() line separator} and if the next
+     * character in comments is not character {@code #} or character {@code !} then
+     * an ASCII {@code #} is written out after that line separator.
```

Comment:
Should something be done for comments ending with \ (backslash) ? It might otherwise suppress the first property assignment that follows.

Reply:
There has been no change in how we deal with this aspect. The existing specification (stated in the `load` method) says:


>     * Properties are processed in terms of lines. There are two
>     * kinds of line, <i>natural lines</i> and <i>logical lines</i>.
>     * A natural line is defined as a line of
>     * characters that is terminated either by a set of line terminator
>     * characters ({@code \n} or {@code \r} or {@code \r\n})
>     * or by the end of the stream. A natural line may be either a blank line,
>     * a comment line, or hold all or some of a key-element pair. A logical
>     * line holds all the data of a key-element pair, which may be spread
>     * out across several adjacent natural lines by escaping
>     * the line terminator sequence with a backslash character
>     * {@code \}.  **Note that a comment line cannot be extended
>     * in this manner;**
(emphasis on that last sentence).
I'll anyway go ahead and add new tests around this to be sure that this works as advertised.

Reply:
Oh - great - thanks - verifying by a test that it also applies to the comment specified by `java.util.Properties.storeDate` would be good.

Reply:
Done. A new test `StoreReproducibilityTest#testBackSlashInStoreDateValue` has been added in the latest updated version of this PR. This test passes. Plus I checked the written out properties file, from these tests, for such values in `java.util.Properties.storeDate` and the content matches what the spec says. 
Just for quick reference - a run of that test case with the "newline-plus-backslash...." system property value (cannot paste that exact string value from that test case because GitHub editor is messing up the special characters) generates output like below:

```
#some user specified comment
#newline-plus-backslash\
#c=d
a=b
```